src/components/TxHistory/TxHistory.js

```diff
@@ -105,57 +101,43 @@ type FundInfo = ?{|
 |}
 
 type Props = {|
-  transactionsInfo: Dict<TransactionInfo>,
   navigation: Navigation,
-  isSyncing: boolean,
-  isOnline: boolean,
-  fetchAccountState: () => Promise<void>,
-  isFetchingAccountState: boolean,
-  updateHistory: () => Promise<void>,
-  lastSyncError: any,
-  tokenBalance: MultiToken,
-  availableAssets: Dict<Token>,
-  isFlawedWallet: boolean,
-  walletMeta: ReturnType<typeof walletMetaSelector>,
+  route: any,
   intl: IntlShape,
 |}
-const TxHistory = ({
-  transactionsInfo,
-  navigation,
-  isSyncing,
-  isOnline,
-  fetchAccountState,
-  isFetchingAccountState,
-  updateHistory,
-  lastSyncError,
-  tokenBalance,
-  availableAssets,
-  isFlawedWallet,
-  walletMeta,
-  intl,
-}: Props) => {
-  // Byron warning banner
 
+const TxHistory = ({navigation, intl}: Props) => {
+  const dispatch = useDispatch()
+
+  const transactionsInfo = useSelector(transactionsInfoSelector)
+  const isSyncing = useSelector(isSynchronizingHistorySelector)
+  const lastSyncError = useSelector(lastHistorySyncErrorSelector)
+  const isOnline = useSelector(isOnlineSelector)
+  const tokenBalance = useSelector(tokenBalanceSelector)
+  const availableAssets = useSelector(availableAssetsSelector)
+  const isFlawedWallet = useSelector(isFlawedWalletSelector)
+  const walletMeta = useSelector(walletMetaSelector)
+  const isFetchingAccountState = useSelector(isFetchingAccountStateSelector)
+  const isWalletInitialized = useSelector(walletIsInitializedSelector)
+
+  // Byron warning banner
   const [showWarning, setShowWarning] = useState<boolean>(isByron(walletMeta.walletImplementationId))
 
   // InsufficientFundsModal (Catalyst)
-
   const [showInsufficientFundsModal, setShowInsufficientFundsModal] = useState<boolean>(false)
 
-  // fetch account state
-
-  useEffect(() => {
-    fetchAccountState()
-  }, [fetchAccountState])
-
   // Catalyst voting registration banner
-
   const canVote = isHaskellShelley(walletMeta.walletImplementationId)
-
   const [showCatalystBanner, setShowCatalystBanner] = useState<boolean>(canVote)
 
   useEffect(() => {
-    const checkCatalystFundInfo = async () => {
+    const run = async () => {
+      // TODO: create actions
+      await dispatch(fetchAccountState())
```

Comment:
prefer more, smaller `useEffect`'s
we want to extract most of them into reusable named effects, like, `useAccountState()` or whatever
and its easier to see when this is appropriate when the `useEffect`'s are super focused
additionally, naming generic things like `run` makes this more difficult